sdk/ml/azure-ai-ml/azure/ai/ml/entities/_indexes/input/_index_data_source.py

```diff
@@ -0,0 +1,234 @@
+        curr_file_path = os.path.dirname(__file__)
+        if ai_search_index_config:
+            ai_search_index_name = ai_search_index_config.ai_search_index_name
+            ai_search_index_import_config = json.dumps({"index_name": ai_search_index_name})
+            git_create_or_update_acs_component = load_component(
+                os.path.join(curr_file_path, "component-configs", "git_create_or_update_acs_index.yml")
+                )
+            rag_job_component: Pipeline =  git_create_or_update_acs_component(
+                embeddings_dataset_name=index_config.output_index_name,
+                git_connection=self.git_connection_id,
+                git_repository=self.git_url,
+                branch_name=self.git_branch_name,
+                data_source_url=index_config.data_source_url,
+                embeddings_model=index_config.embeddings_model,
+                embedding_connection=index_config.aoai_connection_id,
+                chunk_size=index_config.chunk_size,
+                chunk_overlap=index_config.chunk_overlap,
+                input_glob=index_config.input_glob,
+                max_sample_files=index_config.max_sample_files,
+                chunk_prepend_summary=index_config.chunk_prepend_summary,
+                document_path_replacement_regex=index_config.document_path_replacement_regex,
+                embeddings_container=index_config.embeddings_container,
+                acs_connection=ai_search_index_config.ai_search_index_connection_id,
+                acs_config=ai_search_index_import_config
+            )
+            return rag_job_component
+        else:
+            data_to_faiss_component: PipelineComponent = load_component(
+                os.path.join(curr_file_path, "component-configs", "git_to_faiss.yml")
+            )
+            rag_job_component: Pipeline = data_to_faiss_component(  # type: ignore[no-redef]
+                embeddings_dataset_name=index_config.output_index_name,
+                git_connection=self.git_connection_id,
+                git_repository=self.git_url,
+                branch_name=self.git_branch_name,
+                data_source_url=index_config.data_source_url,
+                embeddings_model=index_config.embeddings_model,
+                embedding_connection=index_config.aoai_connection_id,
+                chunk_size=index_config.chunk_size,
+                chunk_overlap=index_config.chunk_overlap,
+                input_glob=index_config.input_glob,
+                max_sample_files=index_config.max_sample_files,
+                chunk_prepend_summary=index_config.chunk_prepend_summary,
+                document_path_replacement_regex=index_config.document_path_replacement_regex,
+                embeddings_container=index_config.embeddings_container,
+            )
+            rag_job_component.properties["azureml.mlIndexAssetName"] = index_config.output_index_name
```

Comment:
We don't need to set such properties for ACS component?

Reply:
hmm, I don't think `_createComponent` for these components are ever used. Let me remove them